src/resource/nonRdfData.ts

```diff
@@ -243,6 +243,36 @@ export async function overwriteFile(
   return Object.assign(blobClone, { internal_resourceInfo: resourceInfo });
 }
 
+/**
+ * @hidden
+ * This function feels unnecessarily complicated, but is required in order to
+ * have Headers according to type definitions in both Node and browser environments.
```

Comment:
Btw, "type definitions" should have nothing to do with this, as they have no runtime effect. I'm guessing that `fetch` does an `instanceof` check, and since the global `Headers` are apparently not equal to the `Headers` from cross-fetch`, it treats it as a plain object.